content/deyemiobaa/building-a-todolist-with-rails.md

```diff
@@ -0,0 +1,354 @@
+3. `Todo`: This is the name of the model we want to create. Rails will automatically pluralize this to create a table named `todos`.
+4. `description`: This is the column name we want our table to have. Columns are usually created with a default string datatype. We could do `clicks:integer` to add a column with an integer datatype instead.
+
+After running this command, you'll see a new file in the `db/migrate` folder. This file contains the migration code that will be used to create the table and the column in our database. The migration file name is in the format `yyyymmddhhmmss_create_todos.rb`. The first part of the file name is the timestamp of when the migration was created. This is used to keep track of the order in which migrations were created. The second part is the name is the kind of migration we want to create.
+
+Run `rails db:migrate` to apply the schema to our local database.
+
+A model file will also be created in the `app/models` folder. This file contains the model class and is used to define the model's attributes and relationships with other models.
+
+## Handling requests within controller methods.
+
+Controllers are responsible for handling requests and sending responses. They are the middleman between the user and the model. They receive requests from the user and send responses back to the user. They also interact with the model to get the data it needs to perform its operations.
+
+Let's create a controller for our todo items. Run `rails generate controller todos`. This will create a controller file in the `app/controllers` folder. It will also create a folder named `todos` in the `app/views` folder. This folder will contain all the views for our todo items.
+
+To display a simple 'Hello World' message, we'll create a new method in the todos controller, and specify routes for it. Open the `app/controllers/todos_controller.rb` file and add the following code.
+
+```ruby
+class TodosController < ApplicationController
+  def index
+  end
+end
+```
+
+This is a simple controller method that will render the index view when a request is made to the `/todos` route. The index view is the default view for a controller. Rails will look for a view with the same name as the controller method. In this case, it will look for a view named `index.html.erb` in the `app/views/todos` folder.
+
+To create this view, create a new file named `index.html.erb` in the `app/views/todos` folder. Add the following code to the file.
+
+```html
+<h1>Hello World</h1>
+```
+
+Then add the following code to the `config/routes.rb` file.
+
+```ruby
+Rails.application.routes.draw do
+  get '/todos', to: 'todos#index'
+end
+```
+
+This will create a route that maps the `/todos` url to the `todos#index` controller method. The `get` method specifies the HTTP verb to use. The `to` option specifies the controller and method to use. The `todos#index` syntax is a shorthand for `controller: 'todos', action: 'index'`.
```

Comment:
```suggestion
This will create a route that maps the `/todos` URL to the `todos#index` controller method. The `get` method specifies the HTTP verb to use. The `to` option specifies the controller and method to use. The `todos#index` syntax is a shorthand for `controller: 'todos', action: 'index'`.
```